server.js

```diff
@@ -1,32 +1,121 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import { Netflix } from "./models/NetflixModel.js";
+import netflixData from "./data/netflix-titles.json";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost:27017/project-mongo";
 mongoose.connect(mongoUrl);
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Seed the database if RESET_DB is set
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Netflix.deleteMany({});
+    netflixData.forEach(async (item) => {
+      await new Netflix(item).save();
+    });
+  };
+  seedDatabase();
+}
+
+// Home route with API documentation
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send({
+    message: "Welcome to the Marias Netflix API!",
+    endpoints: [
+      { path: "/", method: "GET", description: "API documentation" },
+      { path: "/shows", method: "GET", description: "Get all shows" },
+      {
+        path: "/shows/:id",
+        method: "GET",
+        description: "Get a single show by ID",
+      },
+      {
+        path: "/shows/search",
+        method: "GET",
+        description: "Search shows by filters (e.g., title, type, country)",
+      },
+    ],
+  });
+});
+
+// Search shows by filters
+app.get("/shows/search", async (req, res) => {
+  const { title, type, country } = req.query;
+  const query = {};
+
+  if (title) {
+    query.title = new RegExp(title, "i");
+  }
+  if (type) {
+    query.type = type;
+  }
+  if (country) {
+    query.country = country;
+  }
+
+  try {
+    const shows = await Netflix.find(query);
+    res.status(200).json(shows);
+  } catch (err) {
+    res.status(500).json({ error: "Failed to fetch shows", details: err.message });
+  }
+});
+
+// Example stats route
+app.get("/shows/stats", async (req, res) => {
+  try {
+    const count = await Netflix.countDocuments();
+    res.status(200).json({ totalShows: count });
+  } catch (err) {
+    res.status(500).json({ error: "Failed to fetch stats", details: err.message });
+  }
+});
+
+// Get all shows with pagination
+app.get("/shows", async (req, res) => {
```

Comment:
These three could all just be one endpoint called /shows. Then you can use different query params to filter on specific things